client/src/lib/attempt-result.js

```diff
@@ -124,6 +128,47 @@ function mean(values) {
   return Math.round(sum / values.length);
 }
 
+/**
+ * Returns projected average.
+ *
+ * Note that contrarily to other functions in this module, this
+ * function expects a non-padded and incomplete list of attempt
+ * results (without trailing skipped values).
+ *
+ * Projections are defined as follows:
+ *
+ *   - mo3 events: mean of current solves
+ *   - ao5 events:
+ *     - 1-2 solves: mean of current solves
+ *     - 3-4 solves: median of current solves
+ *
+ * When all result attempts are present, the return value is the same
+ * as the usual average.
+ */
+export function projectedAverage(attemptResults, format) {
+  if (attemptResults.length === 0) return SKIPPED_VALUE;
+  
+  if (format.numberOfAttempts === 3) {
+    return meanOfX(attemptResults);
+  }
+  
+  if (format.numberOfAttempts === 5) {
+    if (attemptResults.length < 3) {
+      return meanOfX(attemptResults);
+    }
+    if (attemptResults.length === 3) {
+      const [, x,] = attemptResults.slice().sort(compareAttemptResults);
+      return x;
+    }
+    if (attemptResults.length === 4) {
+      const [, x, y,] = attemptResults.slice().sort(compareAttemptResults);
+      return meanOfX([x, y]);
+    }
```

Comment:
I would still do this for completeness, even if we don't use it directly:

```suggestion
    }
    return averageOf5(attemptResults);
```